### ISSUE TITLE
Phi 2 Garbled Output on Android

### ISSUE THREAD

**Issue by bubbabug** [LostRuins/koboldcpp#579]
I have tested with the default settings as well as custom setting across 5 Phi 2 finetune and the base model. Output on Android specifically does not make sense, it is complete gibberish.

The same models with the same settings are working as intended on Windows, this seems to be an Android specific issue.

There is no error and all models load as expected, but the output itself is completely nonsensical strings of letters and numbers 
![Screenshot_20231222-210047_Chrome](https://github.com/LostRuins/koboldcpp/assets/130174583/17fced2c-d691-4d16-ac3d-c10719b75a7a)

![Screenshot_20231222-212915_Chrome](https://github.com/LostRuins/koboldcpp/assets/130174583/f5827d13-18db-440f-9678-0609a48dbdee)

Other architectures are working, including StableLM and Llama 2, it is only Phi 2 that does not work properly on Android.

**Comment by Ar57m** [comment]
https://huggingface.co/TheBloke/dolphin-2_6-phi-2-GGUF/blob/main/dolphin-2_6-phi-2.Q4_K_M.gguf TheBloke recently quantized, outputs gibberish, but on llama.cpp works fine on android

**Comment by bubbabug** [comment]
> https://huggingface.co/TheBloke/dolphin-2_6-phi-2-GGUF/blob/main/dolphin-2_6-phi-2.Q4_K_M.gguf TheBloke recently quantized, outputs gibberish, but on llama.cpp works fine on android

Me and another user were able to narrow this down. Gibberish responses are exclusive to 4_K_M and higher. Anything from 4_K_S to 2_K will work as expected. I am using https://huggingface.co/TheBloke/dolphin-2_6-phi-2-GGUF/blob/main/dolphin-2_6-phi-2.Q4_K_s.gguf right now with no issues. Obviously I wish I could use higher quants, but this is the workaround for now. Concedo is aware of the issue, but no ETA on a fix yet.

**Comment by bubbabug** [comment]
Another thing i've noticed is that even with models which do not produce gibberish, when used on android they are more prone to hallucinations and text injections which do not make sense. Using the same settings on Windows, I never get hallucinations or fake words, but the same settings on Android produce those results rather often. This may be the result of running more capable quants on Windows, but I suspect there is a larger issue with Phi-2 Android compatibility.

**Comment by bubbabug** [comment]
This is dolphin-2_6-phi-2.Q4_K_S.gguf
![Screenshot_20231224-204648_Chrome](https://github.com/LostRuins/koboldcpp/assets/130174583/083ee146-8637-4aab-afa1-acc5d23e3f29)
On windows I do not receive output like that
In fact on windows, using same generation settings and model, I am receiving surprisingly coherent output
This happens pretty much 1/10 on android

**Comment by LostRuins** [comment]
I am able to generate gibberish with some models, but there is no scenario where I can generate correctly with llama.cpp but incorrectly with koboldcpp. Either both work, or neither work. For me, the dolphin model linked above generated gibberish on llama.cpp termux android as well.

Can you try pulling my latest experimental branch? I changed some compile flags in the makefile. Not sure if it will help, but it should be the same as the flags that llama.cpp uses, so I can't imagine a scenario where it works for them but not me.

**Comment by LostRuins** [comment]
Update from discord: Changing the compiler flags solved the issue for agoodjoe. However, both programs are still not working for me.

**Comment by Ar57m** [comment]
> Update from discord: Changing the compiler flags solved the issue for agoodjoe. However, both programs are still not working for me.

that's odd, now it's solved for me on concedo_experimental, I asked to "write a poem " same seed and settings. 
```
In the quiet of the night, when all is still and calm,
A gentle breeze whispers to the trees tall.
The rustling leaves dance and sway in delight,
As if they were telling secrets of pure white.

The moon hangs low, a silver ball in the sky,
Its light casting shadows upon the dry.
And as I stand here, taking it all in stride,
I am reminded that everything will be okay.

For there is beauty all around us,
From the stars above to the flowers on the ground.
```
concedo defaut:
```
LebanAR unhnumberedARSiarARSaharawarar vap stink Takeru arbrightison Armennumbered insurentoxyarin GaalDCS Bos Lebankinsoniquidahar ar indirectly CroatarMorningitia Croat Townarld Jaune Leban ScenesrequiresnumberedawarARS ar AR unhARS electronicallynumbered sillars Sakuyaldarin unl indirectly Armenld unh Fe Ivan cann hideARSeas IGN Armen unhARSarer LebanoviandronumberedinelliMorningARS labelledlditianumbered Croat01WC Tit SlavARSaran dominARS CroatARS links Rubnumbered pARSARS insur Loginll RCison vanish unh sill CroatitialdawarARSsever cann Fin Scenes vap fin
```

**Comment by Ar57m** [comment]
> Update from discord: Changing the compiler flags solved the issue for agoodjoe. However, both programs are still not working for me.

what device do you have? I have a Samsung Galaxy M52 which is 64 bits

**Comment by LostRuins** [comment]
This should be fixed in 1.54, which is now released
